tests/lib/cli-engine.js

```diff
@@ -254,6 +254,21 @@ describe("CLIEngine", function() {
             });
         });
 
+        // https://github.com/eslint/eslint/issues/5547
+        it("should respect default ignore rules, even with --no-ignore", function() {
+
+            engine = new CLIEngine({
+                cwd: getFixturePath(),
+                ignore: false
+            });
+
+            var report = engine.executeOnText("var bar = foo;", "node_modules/passing.js");
+
+            assert.equal(report.results.length, 1);
+            assert.equal(report.results[0].filePath, getFixturePath("node_modules/passing.js"));
+            assert.equal(report.results[0].messages[0].message, "File ignored because of a matching ignore pattern. Use --no-ignore to override.");
```

Comment:
Good point, we should have two separate messages for default-ignored and regular files.  Although, https://github.com/eslint/eslint/issues/4828#issuecomment-194535273 may make all of this "warning about ignored files" stuff obsolete, since it looks like we will no longer do that.